lib/puppet/functions/lookup.rb

```diff
@@ -65,9 +70,30 @@
 #  - 'unpack_arrays' Set to string value used as a deliminator to join all array values and then split them again. Default is _undef_
 #  - 'merge_hash_arrays' Set to _true_ to merge hashes within arrays. Default is _false_
 #
+# *Converting from Hiera semantics*
+#
+# A hiera _resolution\_type_ is either `priority`, `array` or `hash` and valid behaviors for the
+# `hash` resolution type are `native`, `deep`, or `deeper`. The following table shows how this
+# maps to the merge strategy:
+#
+#     Hiera merge + Config[:merge_behavior]	=> Merge Strategy
+#     ----------------------------------------------------------
+#     priority (default)                       first (default)
+#     array                                    unique
+#     hash + native                            hash
+#     hash + deeper                            deep (with Hashes)
+#     N/A                                      deep (with Arrays)
+#     hash + deep                              N/A
+#
+# There's no 100% correspondence to the Hiera `hash + deeper` since deep will no longer fail when it encounters an
+# `Array` instance.
+#
+# The Hiera `deep` is not implemented. In the Hiera documentation it is considered "largely useless and should be avoided".
```

Comment:
The Hiera documentation needs to either phrase or document that better, and we certainly don't need to repeat that "largely useless and should be avoided" phrasing here. 

If something isn't implemented, why are we documenting it?